std/range/package.d

```diff
@@ -1274,8 +1274,8 @@ pure @safe nothrow unittest
     auto s3 = "string three".byCodeUnit;
     static assert(is(typeof(s3.front) == immutable char));
     auto r2 = s1.chain(s3);
-    // type is promoted
-    static assert(is(typeof(r2.front) == uint));
```

Comment:
If there is a two way requirement, I think its better to temporarily comment this out, then fix DMD, then add this back.

Reply:
Ok.

Reply:
Sequel PR #7043.